### PR TITLE
Node lambda function packaging

### PR DESCRIPTION
**Pull request by neomorphic**
Hi All,

This is a big change to the way we write nodejs lambda functions. In order to get the webpack code working with our files I had to change the way we import modules from the commonjs require to the ES6 import syntax. There were also a number of linting issues that I had to take care of to get the code to build and deploy. Please check all your functions look correct, based on the changes. I have deployed the code to my dev environment and everything seems to have continued working. If we can get this merged, then our uploaded lambda packages will go from +50Mb to ~2Mb each.

### CODE REVIEW

**Review by cgoina — APPROVED** [review]
Looks good to me

File: search/.eslintrc.yml [hunk 1]
```diff
@@ -6,6 +6,10 @@ extends:
   - eslint:recommended
 parserOptions:
   ecmaVersion: 12
+  sourceType: module
 rules:
   no-console: off
   radix: off
+  semi: 2
```
Comment (krokicki) [inline search/.eslintrc.yml:13]: Is this a valid option for semi? Looks like this should be "always" or "never":
https://eslint.org/docs/rules/semi

File: search/.eslintrc.yml [hunk 2]
```diff
@@ -6,6 +6,10 @@ extends:
...
+  no-trailing-spaces: 2
```
Comment (krokicki) [inline search/.eslintrc.yml:14]: Should be boolean

File: search/.eslintrc.yml [hunk 3]
```diff
@@ -6,6 +6,10 @@ extends:
...
+  block-spacing: 2
```
Comment (krokicki) [inline search/.eslintrc.yml:15]: Value should be "always" or "never"

**Review by krokicki — COMMENTED** [review]
Not sure about these eslint options.

File: search/package.json [hunk 4]
```diff
@@ -22,9 +21,11 @@
     "mathjs": "^7.2.0",
     "moment": "^2.25.3",
     "stream": "0.0.2",
+    "upng-js": "^2.1.0",
     "uuid": "^8.2.0"
   },
   "devDependencies": {
+    "aws-sdk": "^2.720.0",
```
Comment (cgoina) [inline search/package.json:28]: Will this really work only as a dev dependency?

Comment (neomorphic) [reply search/package.json:28]: yes, the lambda environment provides it by default. See here:

https://stackoverflow.com/questions/60948911/can-aws-sdk-be-a-development-dependency-when-developing-lambdas-with-nodejs

and

https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html

File: search/.eslintrc.yml (hunk 1 above)
Comment (neomorphic) [reply search/.eslintrc.yml]: The numbers are interchangeable with the text values, but I switched it to "error", which is the correct value.

File: search/.eslintrc.yml (hunk 2 above)
Comment (neomorphic) [reply search/.eslintrc.yml]: Also switched to "error"

File: search/.eslintrc.yml (hunk 3 above)
Comment (neomorphic) [reply search/.eslintrc.yml]: Also switched to "error"

Comment (neomorphic) [reply search/.eslintrc.yml]: https://eslint.org/docs/user-guide/configuring#configuring-rules

**Pull request merged**